my-app/src/Map.js

```diff
@@ -0,0 +1,129 @@
+import { useSearchParams } from 'react-router-dom';
+import React, { useState } from 'react';
+import { GoogleMap, useJsApiLoader , Marker} from "@react-google-maps/api";
+import './css/Map.css';
+
+
+
+
+function Map(){
+
+  const [latitude, setLatitude] = useState(0);
+  const [longitude, setLongitude] = useState(0);
+  const [searchParams] = useSearchParams();
+  
+  const nearbySearch = () => {
+    let currentPos = new window.google.maps.LatLng(latitude, longitude);
+    var service = new window.google.maps.places.PlacesService(document.getElementById('map2'));
+    var request = {
+        type: ['restaurant'],
+        keyword: searchParams.get('nationality'),
+        location: currentPos,
+        radius: '5000',
+    };
+
+    service.nearbySearch(request, function(results, status) {
+        console.log("Call Google nearbySearch API")
+        const restaurantList = document.getElementById('nearbyRestaurants');
+        restaurantList.textContent = "";
+        restaurantList.append(document.createElement("br"));
+        restaurantList.append(document.createElement("br"));
+        if (status === window.google.maps.places.PlacesServiceStatus.OK) {
+          console.log("Google nearbySearch API loaded successfully")
+          for (let i = 0; i<4 && i <results.length; i++){
```

Comment:
Very minor: the spacing in this line is not consistent. For example, i<4 and i <results.length